models/inventory/StockTransfer.ts

```diff
@@ -373,6 +421,64 @@ export abstract class StockTransfer extends Transfer {
 
     return invoice;
   }
+
+  async getReturnDoc(): Promise<StockTransfer | null> {
+    if (!this.items?.length) {
+      return null;
+    }
+
+    const docData = this.getValidDict(true, true);
+    const docItems = docData.items as StockTransferItem[];
+    const returnDocItems: StockTransferItem[] = [];
+
+    const returnBalanceItemsQty = await this.fyo.db.getReturnBalanceItemsQty(
+      this.schema.name,
+      this.name!
+    );
+
+    for (const item of docItems) {
+      if (!item.quantity) {
+        continue;
+      }
+
+      let quantity = -1 * item.quantity;
+
+      if (returnBalanceItemsQty) {
+        const balanceItemQty = returnBalanceItemsQty.filter(
```

Comment:
Using an array where an object would've been a better choice leads to you looping over it where ever it's passed.

Instead if `getReturnBalanceItemsQty` returned an object you'd be getting the quantity in `O(1)` instead.

Reply:
corrected